docs/examples/good_practices/wandb_setup/main.py

```diff
@@ -0,0 +1,195 @@
+
+    # Setup Wandb
+    run = wandb.init(
+    # Set the project where this run will be logged
+    project="awesome-wandb-example",
+    name=os.environ.get("SLURM_JOB_ID"),
+    resume="allow", # See https://docs.wandb.ai/guides/runs/resuming
+    # Track hyperparameters and run metadata
+    config={
+        "learning_rate": learning_rate,
+        "epochs": training_epochs,
+        "weight_decay": weight_decay,
+        "batch_size": batch_size,
+    },
+    )
+
+    # Create a model and move it to the GPU.
+    model = resnet18(num_classes=10)
+    model.to(device=device)
+
+    optimizer = torch.optim.AdamW(model.parameters(), lr=learning_rate, weight_decay=weight_decay)
+
+    # Setup CIFAR10
+    num_workers = get_num_workers()
+    dataset_path = Path(os.environ.get("SLURM_TMPDIR", ".")) / "data"
+    train_dataset, valid_dataset, test_dataset = make_datasets(str(dataset_path))
+    train_dataloader = DataLoader(
+        train_dataset,
+        batch_size=batch_size,
+        num_workers=num_workers,
+        shuffle=True,
+    )
+    valid_dataloader = DataLoader(
+        valid_dataset,
+        batch_size=batch_size,
+        num_workers=num_workers,
+        shuffle=False,
+    )
+    test_dataloader = DataLoader(  # NOTE: Not used in this example.
+        test_dataset,
+        batch_size=batch_size,
+        num_workers=num_workers,
+        shuffle=False,
+    )
+
+    # Checkout the "checkpointing and preemption" example for more info!
+    logger.debug("Starting training from scratch.")
+
+    for epoch in range(training_epochs):
+        logger.debug(f"Starting epoch {epoch}/{training_epochs}")
+
+        # Set the model in training mode (important for e.g. BatchNorm and Dropout layers)
+        model.train()
+
+        # NOTE: using a progress bar from tqdm because it's nicer than using `print`.
+        progress_bar = tqdm(
+            total=len(train_dataloader),
+            desc=f"Train epoch {epoch}",
+        )
+
+        # Training loop
+        for batch in train_dataloader:
+            # Move the batch to the GPU before we pass it to the model
+            batch = tuple(item.to(device) for item in batch)
+            x, y = batch
+
+            # Forward pass
+            logits: Tensor = model(x)
+
+            loss = F.cross_entropy(logits, y)
+
+            optimizer.zero_grad()
+            loss.backward()
+            optimizer.step()
+
+            # Calculate some metrics:
+            n_correct_predictions = logits.detach().argmax(-1).eq(y).sum()
+            n_samples = y.shape[0]
+            accuracy = n_correct_predictions / n_samples
+
+            logger.debug(f"Accuracy: {accuracy.item():.2%}")
+            logger.debug(f"Average Loss: {loss.item()}")
+
+            #Log metrics with wandb
+            wandb.log({"train/accuracy": accuracy, "train/loss": loss})
+
+            # Advance the progress bar one step, and update the "postfix" () the progress bar. (nicer than just)
```

Comment:
This comment here was fixed in the base script, just copy-paste it here so the diff doesn't show that line as different.

Reply:
Done in [e688486](https://github.com/mila-iqia/mila-docs/pull/199/commits/e688486afcd7ed2f07209b6e076cabc7d9dc583d)